Apps/W1/HybridGP/app/src/Migration/Items/GPItemMigrator.codeunit.al

```diff
@@ -300,40 +298,64 @@ codeunit 4019 "GP Item Migrator"
         end;
     end;
 
-    local procedure CreateItemBatch(TemplateName: Code[10]): Code[10]
+    local procedure GetCurrentBatchState()
     var
-        ItemJnlBatch: Record "Item Journal Batch";
+        ItemJournalBatch: Record "Item Journal Batch";
+        ItemJournalLine: Record "Item Journal Line";
     begin
-        ItemJnlBatch.Init();
-        ItemJnlBatch."Journal Template Name" := TemplateName;
-        ItemJnlBatch.Name := CreateBatchName();
-        ItemJnlBatch.Description := SimpleInvJnlNameTxt;
-        ItemJnlBatch.Insert();
+        CurrentBatchNumber := ItemJournalBatch.Count();
 
-        exit(ItemJnlBatch.Name);
+        if ItemJournalBatch.FindLast() then begin
+            ItemJournalLine.SetRange("Journal Template Name", ItemJournalBatch."Journal Template Name");
+            ItemJournalLine.SetRange("Journal Batch Name", ItemJournalBatch.Name);
+
+            CurrentBatchLineNo := ItemJournalLine.Count();
+        end;
     end;
 
-    local procedure CreateBatchName(): Code[10]
+    local procedure CreateOrGetItemBatch(TemplateName: Code[10]): Code[10]
     var
-        BatchName: Text;
+        ItemJnlBatch: Record "Item Journal Batch";
+        BatchName: Code[10];
     begin
-        BatchName := Format(CreateGuid());
-        exit(CopyStr(BatchName, 2, 10));
+        if CurrentBatchNumber = 0 then
+            CurrentBatchNumber := 1;
+
+        if CurrentBatchLineNo >= MaxBatchLineCount() then begin
```

Comment:
Is a procedure better than a simple 1000 value here?

Reply:
Yes, this is the way Nikola always wants it.